fabfile/__init__.py

```diff
@@ -59,6 +72,12 @@ def internal_provision_puppet(environment):
     if (os.path.exists(hiera_secure_path)):
         put(hiera_secure_path, '/etc/puppet/puppet-hilary/environments/%s/hiera/common_hiera_secure.json' % environment)
 
+    # puppet run and start agent
+    # has to be run twice to get puppetdb and nagios running correctly
```

Comment:
puppetdb needs information from the puppet run but it's not available to get that information until after the puppet run, ergo ran twice.
Nagios' problem is a minor permissions issue, which I may fix.